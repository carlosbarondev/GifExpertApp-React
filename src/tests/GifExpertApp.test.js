import { shallow } from 'enzyme';

import { GifExpertApp } from "../GifExpertApp";

describe('Pruebas en <GifExpertApp />', () => {

    let wrapper = shallow(<GifExpertApp />);

    beforeEach(() => {
        jest.clearAllMocks();
        wrapper = shallow(<GifExpertApp />);
    });

    test('debe mostrar <GifExpertApp /> correctamente', () => {

        expect(wrapper).toMatchSnapshot();

    });

    test('debe de mostrar una lista de categorias ', () => {

        const categories = ['Dragon Ball', 'One Punch'];

        wrapper = shallow(<GifExpertApp defaultCategories={categories} />);
        
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('GifGrid').length).toBe(categories.length);
    })


})