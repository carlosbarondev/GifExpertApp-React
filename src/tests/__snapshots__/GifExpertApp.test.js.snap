// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifExpertApp /> debe de mostrar una lista de categorias  1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="Dragon Ball"
      key="Dragon Ball"
    />
    <GifGrid
      category="One Punch"
      key="One Punch"
    />
  </ol>
</Fragment>
`;

exports[`Pruebas en <GifExpertApp /> debe mostrar <GifExpertApp /> correctamente 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;
